mongo/integration/unified/client_entity.go

```diff
@@ -133,21 +176,171 @@ func (c *clientEntity) failedEvents() []*event.CommandFailedEvent {
 	return events
 }
 
+func getCurrentTime() int64 {
+	return time.Now().Unix()
+}
+
 func (c *clientEntity) processStartedEvent(_ context.Context, evt *event.CommandStartedEvent) {
 	if c.getRecordEvents() {
-		c.started = append(c.started, evt)
+		if _, ok := c.observedEvents["commandStartedEvent"]; ok {
+			c.started = append(c.started, evt)
+		}
+		eventListIDs, ok := c.storedEvents["CommandStartedEvent"]
```

Comment:
nit: Same comment here about indentation. For small blocks (e.g. looking up and storing in `observedEvents`), the indentation would be fine, but with much larger blocks, it becomes harder to read. Maybe return early if `ok` is false?